app/assets/javascripts/job_applications/job_applications.js

```diff
@@ -0,0 +1,26 @@
+$(function() {
+  $(".up, .down").click(function (event) {
+    var $this = $(this);
+    event.preventDefault();
+    $.post(event.currentTarget.action).success(function () {
+      var row = $this.parents("tr");
+
+      if ($this.hasClass("up")) {
+        var priPrev = parseInt(row.prev().find(".priority").text());
+        if (isNaN(priPrev)) { priPrev = 1 }
+        var priCur = parseInt(row.find(".priority").text());
+        row.find(".priority").text(priPrev);
+        row.prev().find(".priority").text(priCur);
+        row.prev().before(row);
+      }
+      else {
+        var priCur = parseInt(row.find(".priority").text());
+        var priNext = parseInt(row.next().find(".priority").text());
+        if (isNaN(priNext)) { priNext = parseInt(priCur) }
+        row.find(".priority").text(priNext);
+        row.next().find(".priority").text(priCur);
+        row.next().after(row);
+      }
+    });
+  });
```

Comment:
You should factor out the Javascript that's common for after the if-else section.